generator/src/main/java/com/linkedin/pegasus/generator/JavaCodeUtil.java

```diff
@@ -146,7 +157,17 @@ else if (!checker.isPersistent(definedClass))
         }
         else if (definedClass.outer() == null)
         {
-          final File file = new File(targetDirectory, definedClass.fullName().replace('.', File.separatorChar) + ".java");
+          String path;
+          if (generateLowercasePath) {
+            // Create path this way since fullName() has a recursive call.
+            String fullName = definedClass.fullName();
+            String name = definedClass.name();
+            String packageName = fullName.substring(0, fullName.length() - name.length());
```

Comment:
Use definedClass.getPackage().name() instead?

Reply:
Did you read the comment? I'm not usually in this codebase but I saw that there was a recursive call in the way fullName worked and thus did it this way. Obviously creating it with the package and doing tolower there would be faster but i wasn't sure if it would produce the same result. 

thoughts?

Reply:
@karthikbalasub ^

Reply:
Let us keep it this way to be safe.